blue_heron/lib/blue_heron/address.ex

```diff
@@ -0,0 +1,91 @@
+defmodule BlueHeron.Address do
+  @moduledoc """
+  Helper struct for working with BLE Addresses.
+  """
+
+  alias BlueHeron.Address
+
+  defstruct string: "00:00:00:00:00:00",
+            binary: <<00, 00, 00, 00, 00, 00>>,
+            integer: 0x000000000000
+
+  defimpl Inspect, for: Address do
+    def inspect(address, _opts) do
+      "#Address<#{address.string}>"
+    end
+  end
+
+  defimpl String.Chars, for: Address do
+    def to_string(address) do
+      address.string
+    end
+  end
+
+  @doc """
+  Parses an Address from a colon delimited BLE address string
+  Example:
+
+      iex> BlueHeron.Address.from_string("A4:C1:38:A0:49:8B")
+      #Address<A4:C1:38:A0:49:8B>
+
+  """
+  def from_string(
+        <<oct1::binary-2, ":", oct2::binary-2, ":", oct3::binary-2, ":", oct4::binary-2, ":",
+          oct5::binary-2, ":", oct6::binary-2>> = string
+      ) do
+    addr = String.to_integer(IO.iodata_to_binary([oct1, oct2, oct3, oct4, oct5, oct6]), 16)
+
+    %Address{
+      string: string,
+      binary: <<addr::48>>,
+      integer: addr
+    }
+  end
+
+  @doc """
+  Parses an Address from a 48 bit integer
+  Examples:
+
+      iex> BlueHeron.Address.from_integer(0xA4C138A0498B)
+      #Address<A4:C1:38:A0:49:8B>
+      iex> BlueHeron.Address.from_integer(181149785672075)
+      #Address<A4:C1:38:A0:49:8B>
+
+  """
+  def from_integer(addr) when addr <= 0xFFFFFFFFFFFF do
+    <<oct1, oct2, oct3, oct4, oct5, oct6>> = <<addr::48>>
+
+    string = format_iodata([oct1, oct2, oct3, oct4, oct5, oct6])
+
+    %Address{
+      string: string,
+      binary: <<addr::48>>,
+      integer: addr
+    }
+  end
+
+  @doc """
+  Parses an Address from a 48 bit binary
+  Examples:
+
+      iex> BlueHeron.Address.from_binary(<<0xA4, 0xC1, 0x38, 0xA0, 0x49, 0x8B>>)
+      #Address<A4:C1:38:A0:49:8B>
+      iex> BlueHeron.Address.from_binary(<<164, 193, 56, 160, 73, 139>>)
+      #Address<A4:C1:38:A0:49:8B>
+
+  """
+  def from_binary(<<oct1, oct2, oct3, oct4, oct5, oct6>>) do
+    <<addr::48>> = <<oct1, oct2, oct3, oct4, oct5, oct6>>
+    string = format_iodata([oct1, oct2, oct3, oct4, oct5, oct6])
+
+    %Address{
+      string: string,
+      binary: <<addr::48>>,
+      integer: addr
+    }
+  end
```

Comment:
```suggestion
  def from_binary(<<oct1, oct2, oct3, oct4, oct5, oct6>> = binary) do
    <<addr::48>> = binary
    string = format_iodata([oct1, oct2, oct3, oct4, oct5, oct6])

    %Address{
      string: string,
      binary: binary,
      integer: addr
    }
  end
```

This is just a nit to avoid more binary creations and assignments

Reply:
ahh too much copy paste!